src/Npgsql/NpgsqlConnection.cs

```diff
@@ -2054,5 +2059,11 @@ enum ConnectorBindingScope
     /// <returns>A valid password for connecting to the database</returns>
     public delegate string ProvidePasswordCallback(string host, int port, string database, string username);
 
+    /// <summary>
+    /// Represents a method that allows the application to setup a connection with custom commands.
+    /// </summary>
+    /// <param name="connection">Physical connection to the database</param>
+    public delegate ValueTask PhysicalOpenCallback(NpgsqlConnector connection);
```

Comment:
As this is a non-generic ValueTask and this isn't perf-sensitive, I'd go with Task instead - but this is fine too.